### PR TITLE
Use DKMS module instead of buster-backports kernel

### LINKED ISSUE #4: Use dkms module instead of installing a second kernel ?
Hi, I'm just asking why don't you use wireguard dkms module in buster-backport, instead of using buster-backport kernel ?
This way we keep the LTS kernel, we may still need a reboot anyway.
It will need to set YunoHost 4 as minimum version for Buster

### PR DESCRIPTION
**Pull request by tytan652**
Closes #4 

## Problem
- *I think it would be better to use DKMS module instead of buster-backports kernel*

## Solution
- [x] Add wireguard-dkms as dependency
- [x] Edit scripts to set pin-priority for only wireguard packages to prevent buster-backports kernel to install
- [x] Bump YunoHost minimum version to prevent installation on stretch
- [x] Downgrade linux-image-$arch in upgrade script
- [x] Send a mail about the fact that the user run the backports kernel and should reboot to stable one ?
- [x] Remove the backports kernel automaticaly ~or notify the user to do it~

## PR Status
- [x] Code finished.
- [x] Tested with Package_check.
- [x] Fix or enhancement tested (except mail).
- [x] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Package_check results
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/wireguard_ynh%20PR8%20(tituspijean)/)

### CODE REVIEW

File: scripts/install
Comment (tituspijean) [inline scripts/install:80]: Can you explain why it is needed to "explode" the `ynh_install_extra_app_dependencies` helper? If I read its code right, there is already a 995 priority set on the repositories added by it.

Comment (tytan652) [reply scripts/install:80]: I want to pin-priority only wireguard packages, If I don't it will update linux-image-amd64 to the buster-backports version and install the new kernel

Comment (tytan652) [reply scripts/install:80]: When you install wireguard_ynh, the installation of wireguard provoke the update of linux-image-amd64 to the buster-backports version installing the new kernel in the process.
And if you add wireguard-dkms, it will install the linux-header of the buster-backports kernel because of the new linux-image-amd64.
The solution was to set the pin-priority to 995 for only wireguard packages, the only way to set this is to "explode" this herlper function to change ```--package="*"``` to ```--package='wireguard*'```

Comment (tytan652) [reply scripts/install:80]: By the way I think I should add a way to uninstall the buster-backports linux-image-$ARCH by the right one in the upgrade script

**Comment by tytan652** [comment]
By the way I think I should add a way to uninstall the buster-backports linux-image-$ARCH by the right one in the upgrade script

**Comment by tytan652** [comment]
#8 may be needed to be merged before this, to get a working upgrade script and add needed behavior to upgrade to dkms

**Comment by tytan652** [comment]
Should I remove the backports kernel or send a mail about it ?

**Comment by tytan652** [comment]
Note: the backports kernel will never get automatically upgraded

**Comment by tytan652** [comment]
Waiting for review

**Review by tituspijean — CHANGES_REQUESTED** [review]
Looking good! Some remarks from the package check:

1.  `[Upgrade Script] ! You should not use `ynh_package_remove` or `apt-get remove`, use `ynh_remove_app_dependencies` instead`
2. ` ! You should specify a User= directive in the systemd config ! ` >Root, maybe?
3. During install: `wg-quick.target is a disabled or a static unit, not starting it.`
4. `Port 8095 is already closed for IPv4 connections` during removal. Same for opening during install. Same for IPv6.
5. During restore: `Group 'visitors' already has permission 'wireguard.main' enabled`

**Comment by tytan652** [comment]
>     1. `[Upgrade Script] ! You should not use `ynh_package_remove`or`apt-get remove`, use `ynh_remove_app_dependencies` instead`
> 
I remove the backports kernel and it's not considered as a dependency for wireguard_ynh.

>     2. `! You should specify a User= directive in the systemd config !` >Root, maybe?
> 
Not related to this MR, but you can test it on the testing itself
>     3. During install: `wg-quick.target is a disabled or a static unit, not starting it.`
> 
Not related to this MR, I think it's related to the post-install of wireguard package with apt
>     4. `Port 8095 is already closed for IPv4 connections` during removal. Same for opening during install. Same for IPv6.
> 
Not related to this MR, I don't know why, maybe it's done automatically
>     5. During restore: `Group 'visitors' already has permission 'wireguard.main' enabled`
Not related to this MR, remove adding permission on restore script ?

**Comment by tytan652** [comment]
I can do those changes on this MR if you want, it's just not related to the purpose of this one

**Comment by tytan652** [comment]
User=root give this, I will give it a try
```! DO NOT run the app's systemd service as root or www-data ! Use a dedicated system user for this app ! If your app requires administrator priviledges, you should consider adding the user to the sudoers (and restrict the commands it can use !)```

**Comment by tituspijean** [comment]
You are definitely right about these unrelated warnings... sorry about that. Let's do the relevant changes here. I can take care of it though. ;)

**Comment by tytan652** [comment]
I think we shall ignore, any warning related to the firewall.

**Comment by tytan652** [comment]
The only warning concerning this MR is the one related to ynh_package_remove but I need to use it since backports kernel is not a dependency of wirguard_ynh

**Comment by tytan652** [comment]
Don't forget to replace the REPLACEBYYOURAPP in README_fr.md, please

**Comment by tituspijean** [comment]
Oh wait, I'm on it :)
Let me handle the `User=` thing.

**Comment by tytan652** [comment]
> Oh wait, I'm on it :)
> Let me handle the `User=` thing.

**Comment by tytan652** [comment]
Sorry missclick

**Comment by tytan652** [comment]
I just removed the REPLACEBYYOURAPP. I will do nothing else unrelated to this MR in this MR.

**Comment by tytan652** [comment]
I have no issue with those commits but you could add them to testing branch directly, because it's not related to the MR.
But since this will probably be merged, you can let those commit on this MR.

After the merge, will you ask to make this app considered as working and maintained ?

**Comment by tituspijean** [comment]
I agree, those additions could have been made in another PR. I am done with my digression, and I have successfully upgraded my server with your PR. 🎉 

I could not either test the sending out of the mail during upgrade. But it's a minor feature. Let's merge!
I will definitely mark the app as working. It's already maintained, though? ;)

**Pull request merged**
